### PR TITLE
golang: 1.20.7 -> 1.21

### PR DESCRIPTION
**Pull request by srenatus**
Chore: Update Golang used for building OPA. [Release notes](https://tip.golang.org/doc/go1.21)

### CODE REVIEW

**Comment by srenatus** [comment]
```
--- FAIL: TestJSONSerialization (0.00s)
    prometheus_test.go:142: unexpected extra metrics, expected 72, got 96
```
😆 great

**Comment by srenatus** [comment]
> A few previously-internal GC metrics, such as live heap size, are now available.

I guess that's you

File: plugins/rest/rest_test.go [hunk 1]
```diff
@@ -1274,9 +1274,17 @@ func TestClientCert(t *testing.T) {
...
-		if err == nil || !strings.Contains(err.Error(), expectedErrMsg) {
-			t.Fatalf("Expected '%s' error but request succeeded", expectedErrMsg)
+		expectedErrMsg := func(s string) bool {
+			switch {
+			case strings.Contains(s, "tls: unknown certificate authority"):
+			case strings.Contains(s, "tls: bad certificate"):
+			default:
+				return false
+			}
+			return true
```
Comment (johanfylling) [inline plugins/rest/rest_test.go:1284]: Fallthrough not being the default in go still requires me to think twice 😅.

**Review by johanfylling — APPROVED** [review]
LGTM 👍

**Pull request merged**